### ISSUE TITLE
Some functions not covered in no-unbound-this

### ISSUE THREAD

**Issue by ganimomer** [wix/eslint-plugin-lodash#122]
The rule `no-unbound-this` currently only works for collection methods.
Other Lodash methods have function arguments (e.g. the predicate in `dropWhile`, the resolver in `memoize` that may have the same error but are not reported by the rule.
e.g.
```js
const filtered = _.dropWhile(objs, function(obj) {
  return this.f(obj)
});
```
